lib/reservation_creator.rb

```diff
@@ -0,0 +1,26 @@
+require 'date'
+
+module Hotel
+  class ReservationCreator
+    attr_reader :check_in_date, :check_out_date
+    attr_accessor :room_number
+
+    def initialize(input)
+      @check_in_date = Date.parse("#{input[:check_in_date]}")
+
+      @check_out_date = Date.parse("#{input[:check_out_date]}")
+        raise StandardError, "Checkout date cannot be before checkin date" unless input[:check_in_date] < input[:check_out_date]
+      @room_number = input[:room_number].nil? ? [] : input[:room_number]
```

Comment:
This code says: if in the passed-in `input` hash, there is no value defined for the key `:room_number`, then the value of `@room_number` should be assigned to the empty array `[]`. Otherwise it will be assigned the value of `input[:room_number]`.

I think it makes sense that `@room_number` is `input[:room_number]` if it exists, but should it be assigned to an empty array if there was no given room number?